client/src/partials/patient_invoice/patientInvoice.js

```diff
@@ -157,6 +162,17 @@ function PatientInvoiceController(Patients, PatientInvoices, PatientInvoiceForm,
     }
   }
 
+  function handleServiceChange () {
+    vm.Invoice.setService(vm.Invoice.service);
+    updateDescription();
```

Comment:
Significantly reduced dependencies, requests and lines of code. Great work for maintainability.